front/src/applications/editor/tools/routeEdition/tool.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { isEqual } from 'lodash';
+import { GiPathDistance } from 'react-icons/gi';
+import { AiFillSave } from 'react-icons/ai';
+import { GoPlusCircle, GoTrash } from 'react-icons/go';
+import { BiReset } from 'react-icons/bi';
+
+import { WayPointEntity } from 'types';
+import { save } from 'reducers/editor';
+import { ConfirmModal } from 'common/BootstrapSNCF/ModalSNCF/ConfirmModal';
+import { Tool } from '../editorContextTypes';
+import { RouteEditionState } from './types';
+import { Layers, LeftPanel, Messages } from './components';
+import { getEmptyRoute, getRouteEditionState } from './utils';
+
+const RouteEditionTool: Tool<RouteEditionState> = {
+  id: 'route-edition',
+  icon: GiPathDistance,
+  labelTranslationKey: 'Editor.tools.routes-edition.label',
+  requiredLayers: new Set(['buffer_stops', 'detectors']),
+  getInitialState: () => getRouteEditionState(),
+  actions: [
+    [
+      {
+        id: 'save-route',
+        icon: AiFillSave,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.save-route',
+        isDisabled({ state: { isComplete } }) {
+          return !isComplete;
+        },
+        async onClick({ setIsFormSubmited }) {
+          if (setIsFormSubmited) {
+            setIsFormSubmited(true);
+          }
+        },
+      },
+      {
+        id: 'reset-route',
+        icon: BiReset,
+        labelTranslationKey: `Editor.tools.routes-edition.actions.reset-route`,
+        isDisabled({ state: { entity, initialEntity } }) {
+          return isEqual(entity, initialEntity) || isEqual(entity, getEmptyRoute());
+        },
+        onClick({ state: { initialEntity }, setState }) {
+          setState(getRouteEditionState(initialEntity));
+        },
+      },
+    ],
+    [
+      {
+        id: 'new-route',
+        icon: GoPlusCircle,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.new-route',
+        onClick({ setState }) {
+          setState(getRouteEditionState());
+        },
+      },
+    ],
+    [
+      {
+        id: 'delete-route',
+        icon: GoTrash,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.delete-route',
+        // Show button only if we are editing
+        isDisabled({ state: { entity } }) {
+          return !entity?.properties.id;
```

Comment:
The only case I see when we would want to disable the delete would be when creating a new route from scratch. What do you think ?

```suggestion
        isDisabled({ state: { initialEntity } }) {
          return !initialEntity;
```